librz/analysis/block.c

```diff
@@ -839,34 +839,34 @@ static bool automerge_predecessor_successor_cb(ut64 addr, void *user) {
 		return true;
 	}
 	bool found;
-	RzAnalysisBlock *pred = ht_up_find (ctx->predecessors, (ut64)block, &found);
+	RzAnalysisBlock *pred = ht_up_find (ctx->predecessors, HT_PTR(block), &found);
 	if (found) {
 		if (pred) {
 			// only one predecessor found so far, but we are the second so there are multiple now
-			ht_up_update (ctx->predecessors, (ut64) block, NULL);
+			ht_up_update (ctx->predecessors, HT_PTR(block), NULL);
 		} // else: already found multiple predecessors, nothing to do
 	} else {
 		// no predecessor found yet, this is the only one until now
-		ht_up_insert (ctx->predecessors, (ut64) block, ctx->cur_pred);
+		ht_up_insert (ctx->predecessors, HT_PTR(block), ctx->cur_pred);
 	}
 	return true;
 }
 
 static bool automerge_get_predecessors_cb(void *user, const ut64 k, const void *v) {
 	AutomergeCtx *ctx = user;
-	const RzAnalysisFunction *fcn = (const RzAnalysisFunction *)k;
+	const RzAnalysisFunction *fcn = (const RzAnalysisFunction *)(size_t)k;
```

Comment:
why that `(size_t)` is needed? k is already 64 bits

Reply:
On ARMv7 the pointer size will be 32bit, thus it will lead to the warning "casting pointer from integer of different size" or something like that.